src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -95,14 +101,51 @@ public TimeSpan GetAnalyzerExecutionTime(DiagnosticAnalyzer analyzer)
             }
         }
 
-        internal ImmutableArray<DiagnosticAnalyzer> GetPendingAnalyzers(ImmutableArray<DiagnosticAnalyzer> analyzers)
+        private HashSet<DiagnosticAnalyzer>? GetCompletedAnalyzersForFile_NoLock(SourceOrAdditionalFile filterFile, bool syntax)
+        {
```

Comment:
_NoLock methods should have a Contract call that asserts Monitor.IsEntered(_gate)

Reply:
> _NoLock methods should have a Contract call that asserts Monitor.IsEntered(_gate)

That is not the case throughout the compiler/analyzer layer. I'll create a separate follow-up PR to add this assert for the entire analyzer layer.